### DISCUSSION TITLE
Model Comparison and Specification of Priors

### DISCUSSION THREAD

**Discussion by xprockox** [easystats/bayestestR#461] [Q&A]
Hi there,

In the vignette, there's a really nice tutorial on [using Bayes Factors for model comparison](https://easystats.github.io/bayestestR/articles/bayes_factors.html#comparing-models-using-bayes-factors). The sample provided constructs a few variations of models to compare:

># intercept only model
>m0 <- brm(Sepal.Length ~ 1, data = iris, 
          save_pars = save_pars(all = TRUE),  refresh = 0, backend = "rstan")
>
># main effects models
>m1 <- brm(Sepal.Length ~ Petal.Length, data = iris, 
          save_pars = save_pars(all = TRUE), refresh = 0, backend = "rstan")
>
>m2 <- brm(Sepal.Length ~ Species, data = iris,
          save_pars = save_pars(all = TRUE), refresh = 0, backend = "rstan")
>
># additive main effects model
>m3 <- brm(Sepal.Length ~ Species + Petal.Length, data = iris,
          save_pars = save_pars(all = TRUE), refresh = 0, backend = "rstan")
>
># full model
>m4 <- brm(Sepal.Length ~ Species * Petal.Length, data = iris,
          save_pars = save_pars(all = TRUE), refresh = 0, backend = "rstan")

Then, models can be easily compared using bayestestR::bayesfactor_models().

However, from what I've read, computing Bayes Factors requires that models specify informative priors, which doesn't appear to have been done in these models. Is the specification of priors only required for testing model parameters? Or am I missing something here?

Any feedback would be greatly appreciated!

**Comment by mattansb** [comment]
Oh my god, you are 110% right! 😨😨😨😨😨

I will fix the vignette!!!

**Reply by mattansb** [reply]
Fixed!!

**Reply by xprockox** [reply]
Excellent, thank you!

**Comment by bwiernik** [comment]
![image](https://user-images.githubusercontent.com/4773225/132550448-8956c777-1822-419c-b02c-c9580549a15d.png)
